src/shared/components/oncoprint/DataUtils.ts

```diff
@@ -283,6 +283,9 @@ export function fillHeatmapTrackDatum<T extends IBaseHeatmapTrackDatum, K extend
 ) {
     trackDatum[featureKey] = featureId;
     trackDatum.study_id = case_.studyId;
+    if (data) {
```

Comment:
this mutates the array of items passed in as argument which is a bit of a no-no.   you might want to use filter instead to produce a new array (elements will not be cloned)

Reply:
Ok, code updated to use _.filter and capture of filtered array in a a new variable.

Reply:
Reading your comment again, I am sure that this is also the way this method works in the current master; the `trackDatum` that is passed in is returned after modification. Maybe we have to consult with @adamabeshouse to find out whether this is a thing that we need to fix?